hera_pspec/uvpspec.py

```diff
@@ -2180,35 +2180,26 @@ def combine_uvpspec(uvps, merge_history=True, verbose=True):
         if store_stats:
             for stat in stored_stats:
                 u.stats_array[stat][i] = np.empty((Nblpairts, spw[3], Npols), np.complex128)
-        # Set frequencies and delays
-        spw_Nfreqs = spw[2]
-        spw_Ndlys = spw[3]
-        spw_freqs = np.linspace(*spw[:3], endpoint=False)
-        spw_dlys = np.fft.fftshift(
-                        np.fft.fftfreq(spw_Ndlys,
-                                       np.median(np.diff(spw_freqs)) ) )
-        u.spw_freq_array.extend(np.ones(spw_Nfreqs, np.int32) * i)
-        u.spw_dly_array.extend(np.ones(spw_Ndlys, np.int32) * i)
-        u.spw_array.extend([i])
-        u.freq_array.extend(spw_freqs)
-        u.dly_array.extend(spw_dlys)
-
-    # Convert to numpy arrays
-    u.spw_array = np.array(u.spw_array)
-    u.spw_freq_array = np.array(u.spw_freq_array)
-    u.spw_dly_array = np.array(u.spw_dly_array)
-    u.freq_array = np.array(u.freq_array)
-    u.dly_array = np.array(u.dly_array)
+        # Set frequencies and delays: if concat_ax == 'spw' this is changed below
```

Comment:
Seems we could move this block outside the loop?

Reply:
ah yes, sure thing